src/host/tcp.h

```diff
@@ -31,7 +31,7 @@ namespace asynchost
     friend class close_ptr<TCPImpl>;
 
     static constexpr int backlog = 128;
-    static constexpr size_t read_size = 1024;
+    static constexpr size_t max_read_size = 16384;
```

Comment:
This reduces the output spam for large messages (reduces the number of reads by a factor of 16 so we), and should also improve performance for them. But this may be at the cost of performance for smaller messages, and was a deadend in tracking down the actual failure so should perhaps be removed.

Reply:
Should this be a comment in the code as well?

Reply:
It seems like this doesn't actually have a noticeable impact on performance either way, so I think we can just ignore this for now (and hopefully remember it as a potentially bad choice if it comes back to bite us in future?).

I think it should actually be possible to just keep a single buffer in this object, rather than reallocating on every request, but I'll also leave that investigation on the backburner atm.